deploy/awstools/awstools.py

```diff
@@ -153,13 +158,53 @@ def construct_instance_market_options(instancemarket, spotinterruptionbehavior,
     else:
         assert False, "INVALID INSTANCE MARKET TYPE."
 
-def launch_instances(instancetype, count, instancemarket, spotinterruptionbehavior, spotmaxprice, blockdevices=None, tags=None, randomsubnet=False, user_data_file=None):
-    """ Launch count instances of type instancetype, optionally with additional
-    block devices mappings and instance tags
-
-         This will launch instances in avail zone 0, then once capacity runs out, zone 1, then zone 2, etc.
+def launch_instances(instancetype, count, instancemarket, spotinterruptionbehavior, spotmaxprice, blockdevices=None,
+                     tags=None, randomsubnet=False, user_data_file=None, timeout=timedelta(), additive=True):
+    """ Launch `count` instances of type `instancetype`
+
+    Using `instancemarket`, `spotinerruptionbehavior` and `spotmaxprice` to define instance market conditions
+    (see also: construct_market_conditions)
+
+    This will launch instances in avail zone 0, then once capacity runs out, zone 1, then zone 2, etc.
+    The ordering of availablility zones can be randomized by passing`randomsubnet=True`
+
+    Parameters
+    ----------
+    instancetype : str
+        String acceptable by `boto3.ec2.create_instances()` `InstanceType` parameter
+    count : int
+        The number of instances to launch
+    instancemarket
+    spotinterruptionbehavior
+    spotmaxprice
+    blockdevices
+    tags : dict of tag names to string values, default=None
+        Dict of tags
+    randomsubnet : bool, default=False
+        If true, subnets will be chosen randomly instead of starting from 0 and proceeding incrementally.
+    user_data_file : str, default=None
+        Path to readable file.  Contents of file are passed as `UserData` to AWS
+    timeout : datetime.timedelta, default=timedelta() (immediate timeout after attempting all subnets)
+        `timedelta` object representing how long we should continue to try asking for instances
+    additive : bool, default=True
+        When true, create `count` instances, regardless of whether any already exist. When False, only
+        create instances until there are `count` total instances that match `tags` and `instancetype`
+        If `tags` are not passed, `additive` must be `True` or `UserError` is thrown.
```

Comment:
Nit: Shouldn't this be `ValueError`?

Reply:
Yes.  This is a good catch.  I typed the docs before actually looking at the builtin Exception types and remembering that UserError isn't one of them...
```suggestion
        If `tags` are not passed, `additive` must be `True` or `ValueError` is thrown.
```